xontrib/vox.py

```diff
@@ -75,121 +63,113 @@ def parser():
-            "--remove",
-            dest="remove",
-            default=False,
-            action="store_true",
-            help="Remove the virtual environment after leaving it.",
-        )
-        subparsers.add_parser(
-            "list",
-            aliases=["ls"],
-            help=("List environments available in " "$VIRTUALENV_HOME"),
-        )
-        remove = subparsers.add_parser(
-            "remove", aliases=["rm", "delete", "del"], help="Remove virtual environment"
-        )
-        remove.add_argument(
-            "names",
-            metavar="ENV",
-            nargs="+",
-            help=(
-                "The environments to remove. ENV can be "
-                "either a name from the venvs shown by vox"
-                "list or the path to an arbitrary venv"
-            ),
-        )
-        subparsers.add_parser("help", help="Show this help message")
+        parser.add_command(self.activate, aliases=["workon", "enter"])
+        parser.add_command(self.deactivate, aliases=["exit"])
+        parser.add_command(self.list, aliases=["ls"])
+        parser.add_command(self.remove, aliases=["rm", "delete", "del"])
+        parser.add_command(self.help)
         return parser
 
-    parser = lazyasd.LazyObject(parser, locals(), "parser")
-
-    aliases = {
-        "create": "new",
-        "workon": "activate",
-        "enter": "activate",
-        "exit": "deactivate",
-        "ls": "list",
-        "rm": "remove",
-        "delete": "remove",
-        "del": "remove",
-    }
-
-    def __init__(self):
-        self.vox = voxapi.Vox()
-
-    def __call__(self, args, stdin=None):
-        """Call the right handler method for a given command."""
-
-        args = self.parser.parse_args(args)
-        cmd = self.aliases.get(args.command, args.command)
-        if cmd is None:
-            self.parser.print_usage()
-        else:
-            getattr(self, "cmd_" + cmd)(args, stdin)
-
-    def cmd_new(self, args, stdin=None):
-        """Create a virtual environment in $VIRTUALENV_HOME with python3's ``venv``."""
+    def new(
+        self,
+        name: Annotated[str, Arg(metavar="ENV")],
+        interpreter: Annotated[
+            str,
+            Arg("-p", "--interpreter"),
+        ] = None,
+        system_site_packages: Annotated[
+            bool,
+            Arg("--system-site-packages", "--ssp", action="store_true"),
+        ] = False,
+        symlinks: bool = False,
+        with_pip: bool = True,
+        activate: Annotated[
+            bool,
+            Arg("-a", "--activate", action="store_true"),
+        ] = False,
+    ):
+        """
+            wraps around vox.create and vox.activate
+        Parameters
+        ----------
+        name : str
+            Virtual environment name
+        interpreter: str
+            Python interpreter used to create the virtual environment.
+            Can be configured via the $VOX_DEFAULT_INTERPRETER environment variable.
```

Comment:
For some reason I don't see `--interpreter`'s description in the tab completion :(

Reply:
fixed it. I will update it a new PR. just waiting for #4379

Reply:
Hi is this fixed in #4437?

Reply:
Yes. Please make sure to run the aliases locally though.